### DISCUSSION TITLE
Disappointed whilist trying to set up a development instance to contribute.

### DISCUSSION THREAD

**Discussion by configari** [LBPUnion/ProjectLighthouse#684] [Q&A]
Spent 5 hours trying to set up with and without docker got errors and couldn't succeed and apparently the developers wouldn't help basically I'm disappointed. **I'm asking the community if they have an answer**.

**Comment by sudokoko** [comment]
I don't see the problem with the developers not providing support directly. According to the license (AGPLv3), the software is provided as-is with no warranty. We aren't obligated to provide support. What you _can_ do is open an issue and provide the errors you're getting. What would also help is if you provided the errors you're receiving in this thread and maybe I can see about pointing you in the right direction.

**Reply by maximmax42** [reply]
Posting some errors on behalf of @configari:
- `docker-compose up` throws:
```
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.api.depends_on contains an invalid type, it should be an array
services.gameserver.depends_on contains an invalid type, it should be an array
services.website.depends_on contains an invalid type, it should be an array
```
- Trying to build from Dockerfile (`docker build - < Dockerfile`) results in the error on line 14:
```
------
 > [build 6/8] RUN dotnet restore:
#0 0.662 MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.
------
Dockerfile:13
--------------------
  11 |         done;
  12 |     
  13 | >>> RUN dotnet restore
  14 |     
  15 |     COPY . .
--------------------
```
Because apparently `COPY *.sln ./` and `COPY **/*.csproj ./` do nothing.
- Trying to just run it via Visual Studio/dotnet - seems to work, but also not, CSS on the website never works.

**Reply by sudokoko** [reply]
I just talked with our Lead Engineer and he said you may have cloned the repository wrong. Try re-cloning it and try again.

**Reply by maximmax42** [reply]
Did `git clone https://github.com/LBPUnion/ProjectLighthouse`. Nope, same results.

**Reply by daggintosh** [reply]
CSS is located in ProjectLighthouse/StaticFiles and for some reason isn't getting copied to the server directory, for now the workaround is to copy the StaticFiles folder to StaticFiles in the server directory.

**Reply by TorutheRedFox** [reply]
> CSS is located in ProjectLighthouse/StaticFiles and for some reason isn't getting copied to the server directory, for now the workaround is to copy the StaticFiles folder to StaticFiles in the server directory.

that does make it a little less of a hassle to upgrade your instance if you're using custom css though

**Reply by Slendy** [reply]
Can you post the results of `docker --version` and `docker-compose --version`? Also, I believe the correct usage to build an image using a dockerfile is `docker build .`. If you aren't in the source directory then you would specify the folder that the Dockerfile is in.

**Reply by Slendy** [reply]
> > CSS is located in ProjectLighthouse/StaticFiles and for some reason isn't getting copied to the server directory, for now the workaround is to copy the StaticFiles folder to StaticFiles in the server directory.
> 
> that does make it a little less of a hassle to upgrade your instance if you're using custom css though

In most cases, if you aren't using docker you can create a symbolic link in the runtime folder of the StaticFiles in the source

**Reply by maximmax42** [reply]
> Can you post the results of `docker --version` and `docker-compose --version`? 

Docker version 23.0.1, build a5ee5b1
docker-compose version 1.25.0, build unknown

> Also, I believe the correct usage to build an image using a dockerfile is `docker build .`.

Ah. Yep, that's on me. This does indeed work. Still, docker-compose file is apparently invalid.

**Reply by Slendy** [reply]
I believe the issue is that you're running docker-compose version 1.25.0 and the compose file uses the long-form version of `depends_on` which was only introduced in version [1.29.0](https://docs.docker.com/compose/release-notes/#1290). Looking at the docker documentation I don't see a minimum compose version to prevent parsing errors but for now, you should just have to update your docker-compose.

**Reply by Zaprit** [reply]
~~what problems were you having with the docker-compose file~~ my bad, apparently i couldn't see the show more replies. I have an instance using the docker-compose with no problems. It is however not an out of box kind of a thing to setup due to the config system not playing that nice with docker, my main advice would be changing the mysql and redis lines in lighthouse.config.yml in the data folder to use db and redis instead of 127.0.0.1 or localhost, this make it use the containers ip rather than relying on the host to forward the connections

**Reply by Zaprit** [reply]
> I believe the issue is that you're running docker-compose version 1.25.0 and the compose file uses the long-form version of `depends_on` which was only introduced in version [1.29.0](https://docs.docker.com/compose/release-notes/#1290). Looking at the docker documentation I don't see a minimum compose version to prevent parsing errors but for now, you should just have to update your docker-compose.

the out of date docker-compose probably isn't helping. for the record my server is on `Docker Compose version 2.16.0`

**Reply by sudokoko** [reply]
```
sudo apt-get update
sudo apt-get install docker-compose-plugin
```
:wink:

**Reply by maximmax42** [reply]
Well. Today I learned that I have both `docker-compose` and `docker compose`, and they're different versions. Former being 1.25, latter being 2.16.

**Reply by maximmax42** [reply]
Okay, built and launched. New problem now - nothing can connect to the database, even tho it's running.

**Reply by Slendy** [reply]
This is a side effect of having everything in docker, basically everything is segregated to it’s own network so if you haven’t already rename the `lighthouse.yml.configme` to `lighthouse.yml` and for the db and redis connection strings, the ip should be `db` or `redis` instead of `127.0.0.1` or `localhost`. The same applies for the listenUrls and the addresses will coincide with the name of the service in the compose file

**Reply by maximmax42** [reply]
Okay, website works. Now, I assume we also have to set up nginx as well? Using `localhost:16001` instead of `localhost:10060/LITTLEBIGPLANETPS3_XML` won't work? Because currently `localhost:16001` spits out 404. And so does `localhost:16002`.

**Reply by Slendy** [reply]
Yeah, by default none of the containers should be accessible from outside your local machine unless you’ve modified your firewall settings. There is an example nginx config in the repository that should work for reverse proxying though.

**Reply by maximmax42** [reply]
Well, I can't access server and api on my local machine either, unlike the website. Also, don't I have to set up nginx in docker too, or should that be done on the host?

**Reply by maximmax42** [reply]
Oh, hold on. `localhost:16001/LITTLEBIGPLANETPS3_XML/status` returns 200. Wait but... If I set up proxy as
```
location /LITTLEBIGPLANETPS3_XML/ {
                proxy_http_version 1.1;
                proxy_pass http://127.0.0.1:10061;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }
```
wouldn't that mean that `example.com/LITTLEBIGPLANETPS3_XML` equals to `localhost:16001` and not to `localhost:16001/LITTLEBIGPLANETPS3_XML`? So that means if I need to get the patching url, it'd be `example.com/LITTLEBIGPLANETPS3_XML/LITTLEBIGPLANETPS3_XML`? Or am I confusing myself?

**Comment by Zaprit** [comment]
The ports from the lighthouse containers are forwarded by default, you can either set up nginx in a container or on the host, it’s up to you, either should work

On 21 Feb 2023, at 18:36, Max ***@***.***> wrote:


Well, I can't access server and api on my local machine either, unlike the website. Also, don't I have to set up nginx in docker too, or should that be done on the host?
